ibis/tests/all/test_outer_join.py

```diff
@@ -0,0 +1,41 @@
+import os
```

Comment:
Thanks for writing a unit test!

The way that we typically write unit tests is using a slightly different style, namely that of [`pytest`](https://docs.pytest.org/en/latest/). `pytest` uses a more functional style and avoids mixing up fixtures and test cases. Another nice thing is that it has a lot of builtin fixtures for common tasks like generating a temporary file.

Do you want to try rewriting this test in that style?

Here's a skeleton to get you started:

```python
@pytest.fixture
def client(tmpfile):  # tmpfile is passed in by pytest when the test runs
    con = ibis.sqlite.connect(str(tmpfile), create=True)
    # setup tables ...
    return con


@pytest.fixture
def left(client):
    return client.table('left_table')


@pytest.fixture
def right(client):
    return client.table('right_table')


def test_outer_join(left, right):
    ...
```

Another thing is that you should put this test in `ibis/sql/sqlite/tests/test_functions.py`.

After addressing those two things, this PR will be good to go!

Reply:
Sounds great! I'm new to `pytest`, so thanks for the skeleton. I'll rework it and get back to you.

Reply:
@cpcloud, I re-did the my test in the pytest fashion. Instead of doing so in SQLite, however, I did it in the higher-level `test_sqlalchemy.py`, because the issue wasn't specific to SQLite but to anything based on `sqlalchemy`. Along the way I discovered some issues with the MockConnection and the `test_joins` method, so I fixed and refactored those. 

These changes are in commit:  
scottcode:fix_full_outer_join  29b4a5106a3043bc

I noticed some changes in the upstream master since my first version of pull request. Do I need to make a new rebased pull request branch, or can you rebase when accepting the request? I haven't encountered this situation before in GitHub, so not sure what the convention is.